infra-as-code/bicep/orchestration/hubPeeredSpoke/hubPeeredSpoke.bicep

```diff
@@ -0,0 +1,206 @@
+  baseSuffixSubscription: '${parLocation}-${uniqueString(parLocation, parTopLevelManagementGroupPrefix)}-sub'
+  baseSuffixResourceGroup: '${parLocation}-${uniqueString(parLocation, parTopLevelManagementGroupPrefix)}-rg'
+}
+
+var varModuleDeploymentNames = {
+  modSubscriptionPlacement: take('${varDeploymentNameWrappers.basePrefix}-modSubscriptionPlacement-${varTargetManagementGroupID}-${varDeploymentNameWrappers.baseSuffixManagementGroup}', 64)
+  modResourceGroup: take('${varDeploymentNameWrappers.basePrefix}-modResourceGroup-${varDeploymentNameWrappers.baseSuffixSubscription}', 64)
+  modSpokeNetworking: take('${varDeploymentNameWrappers.basePrefix}-modSpokeNetworking-${varDeploymentNameWrappers.baseSuffixResourceGroup}', 61)
+  modSpokePeeringToHub: take('${varDeploymentNameWrappers.basePrefix}-modVnetPeering-ToHub-${varDeploymentNameWrappers.baseSuffixResourceGroup}', 61)
+  modSpokePeeringFromHub: take('${varDeploymentNameWrappers.basePrefix}-modVnetPeering-FromHub-${varDeploymentNameWrappers.baseSuffixResourceGroup}', 61)
+  modVnetPeeringVwan: take('${varDeploymentNameWrappers.basePrefix}-modVnetPeeringVwan-${varDeploymentNameWrappers.baseSuffixResourceGroup}', 61)
+}
+
+var varHubVirtualNetworkName = (!empty(parHubVirtualNetworkID) && contains(parHubVirtualNetworkID, '/providers/Microsoft.Network/virtualNetworks/') ? split(parHubVirtualNetworkID, '/')[8] : '' )
+
+var varHubVirtualNetworkResourceGroup = (!empty(parHubVirtualNetworkID) && contains(parHubVirtualNetworkID, '/providers/Microsoft.Network/virtualNetworks/') ? split(parHubVirtualNetworkID, '/')[4] : '' )
+
+var varHubVirtualNetworkSubscriptionId = (!empty(parHubVirtualNetworkID) && contains(parHubVirtualNetworkID, '/providers/Microsoft.Network/virtualNetworks/') ? split(parHubVirtualNetworkID, '/')[2] : '' )
+
+var varNextHopIPAddress = (!empty(parHubVirtualNetworkID) && contains(parHubVirtualNetworkID, '/providers/Microsoft.Network/virtualNetworks/') ? parNextHopIPAddress : '' )
+
+var varVirtualHubResourceId = (!empty(parHubVirtualNetworkID) && contains(parHubVirtualNetworkID, '/providers/Microsoft.Network/virtualHubs/') ? parHubVirtualNetworkID : '' )
+
+var varVirtualHubResourceGroup = (!empty(parHubVirtualNetworkID) && contains(parHubVirtualNetworkID, '/providers/Microsoft.Network/virtualHubs/') ? split(parHubVirtualNetworkID, '/')[4] : '' )
+
+var varVirtualHubSubscriptionId = (!empty(parHubVirtualNetworkID) && contains(parHubVirtualNetworkID, '/providers/Microsoft.Network/virtualHubs/') ? split(parHubVirtualNetworkID, '/')[2] : '' )
+
+// **Scope**
+targetScope = 'managementGroup'
+
+// **Modules**
+// Module - Customer Usage Attribution - Telemtry
+module modCustomerUsageAttribution '../../CRML/customerUsageAttribution/cuaIdManagementGroup.bicep' = if (!parTelemetryOptOut) {
+  scope: managementGroup(parTopLevelManagementGroupPrefix)
+  name: 'pid-${varCuaid}-${uniqueString(parLocation)}'
```

Comment:
Need to make this more unique so it is not the same for every spoke VNet.

Maybe add the spoke subscription ID into the unique string piece?

Reply:
Updated in newest commit